File: MAHAds/mah-ads/src/main/java/com/mobapphome/mahads/mahfragments/MAHDialogFragment.java
package com.mobapphome.mahads.mahfragments;

import android.support.v4.app.DialogFragment;
import android.support.v4.app.FragmentActivity;

import com.mobapphome.mahads.mahfragments.MAHFragmentExeption;

/**
 * Created by settar on 12/20/16.
 */


public class MAHDialogFragment extends DialogFragment {

    public FragmentActivity getActivityMAH() throws MAHFragmentExeption {
        if (getActivity() != null) {
            return getActivity();
        } else {
            throw new MAHFragmentExeption("MAHFragment getActivity() method returns null");
        }
    }
}

File: MAHAds/mah-ads/src/main/java/com/mobapphome/mahads/mahfragments/MAHDialogFragment.kt
package com.mobapphome.mahads.mahfragments

import android.support.v4.app.DialogFragment
import android.support.v4.app.FragmentActivity

import com.mobapphome.mahads.mahfragments.MAHFragmentExeption

/**
 * Created by settar on 12/20/16.
 */


open class MAHDialogFragment : DialogFragment() {

    val activityMAH: FragmentActivity
        @Throws(MAHFragmentExeption::class)
        get() {
            if (activity != null) {
                return activity
            } else {
                throw MAHFragmentExeption("MAHFragment getActivity() method returns null")
            }
        }
}